### PR TITLE
require zipped data if possible

### PR DESCRIPTION
**Pull request by bernt-matthias**
see also https://github.com/galaxyproject/tools-iuc/issues/2312

### CODE REVIEW

File: docs/best_practices/tool_xml.rst [hunk 1]
```diff
@@ -162,6 +162,12 @@ warning messages appear). See `planemo docs <https://planemo.readthedocs.io/en/l
 When tools contain output filters, tests should be included that verify
 this filtering occurs. See `planemo docs <https://planemo.readthedocs.io/en/latest/writing_how_do_i.html#test-output-filters-work>`__ for more information.
 
+Data parameters
+---------------
+
+If a (gz)ipped data type is available (e.g. ``fasta.gz`` or ``fastq.gz``) then
+data parameters should accept both the zipped and the unzipper data type.
```
Comment (abretaud) [inline docs/best_practices/tool_xml.rst]:
```suggestion
data parameters should accept both the zipped and the unzipped data type.
```

File: docs/best_practices/tool_xml.rst [hunk 2]
```diff
@@ -166,7 +166,7 @@ Data parameters
 ---------------
 
 If a (gz)ipped data type is available (e.g. ``fasta.gz`` or ``fastq.gz``) then
```
Comment (peterjc) [inline docs/best_practices/tool_xml.rst]: Since you are editing this paragraph, should that be ``(g)zipped`` rather than ``(gz)ipped``?

**Review by peterjc — APPROVED** [review]

**Comment by peterjc** [comment]
I'd have put "typo" in the PR title, but looks good to me. Thanks!

File: docs/best_practices/tool_xml.rst [hunk 3]
```diff
@@ -162,6 +162,12 @@ warning messages appear). See `planemo docs <https://planemo.readthedocs.io/en/l
 When tools contain output filters, tests should be included that verify
 this filtering occurs. See `planemo docs <https://planemo.readthedocs.io/en/latest/writing_how_do_i.html#test-output-filters-work>`__ for more information.
 
+Data parameters
+---------------
+
+If a (g)zipped data type is available (e.g. ``fasta.gz`` or ``fastq.gz``) then
+data parameters should accept both the zipped and the unzipped data type.
```
Comment (nsoranzo) [inline docs/best_practices/tool_xml.rst]:
```suggestion
If a compressed version of a datatype (e.g. ``fasta.gz`` or ``fastqsanger.bz2``)
is supported by Galaxy, then data parameters should accept both the compressed
and the uncompressed datatypes. The tool should internally decompress the
dataset if the underlying tool cannot handle the compressed formats natively.
```

**Comment by peterjc** [comment]
A more dramatic rewrite, but Nicola's version is good too.

File: docs/best_practices/tool_xml.rst (hunk 3 above)
Comment (bernt-matthias) [reply docs/best_practices/tool_xml.rst]: > supported by the underlying tool

this is to weak in my opinion. I would prefer to let the Galaxy tool unzip the data if the underlying tool does not support zipped files natively. Advantage is that this save a lot of space in user histories.

Comment (peterjc) [reply docs/best_practices/tool_xml.rst]: Yes, but that's a galaxy change not a tool change

Comment (bernt-matthias) [reply docs/best_practices/tool_xml.rst]: I have something like this in mind: 

- https://github.com/galaxyproject/tools-iuc/pull/3542/files
- https://github.com/galaxyproject/tools-iuc/pull/3549/files
- https://github.com/galaxyproject/tools-iuc/pull/3548/files

Comment (peterjc) [reply docs/best_practices/tool_xml.rst]: Sorry, I miss read that as preferring the Galaxy framework do this (which is/was my own view).

The examples make it clear you are advocating extending the tool wrapper to do decompression when the tool itself doesn't.

Comment (nsoranzo) [reply docs/best_practices/tool_xml.rst]: > this is to weak in my opinion. I would prefer to let the Galaxy tool unzip the data if the underlying tool does not support zipped files natively. Advantage is that this save a lot of space in user histories.

I've updated my suggestion above accordingly.

**Review by bgruening — APPROVED** [review]

**Pull request merged**